src/vs/platform/telemetry/common/telemetryUtils.ts

```diff
@@ -113,9 +113,36 @@ export function configurationTelemetry(telemetryService: ITelemetryService, conf
  * @returns false - telemetry is completely disabled, true - telemetry is logged locally, but may not be sent
  */
 export function supportsTelemetry(productService: IProductService, environmentService: IEnvironmentService): boolean {
+	// If it's OSS and telemetry isn't disabled via the CLI we will allow it for logging only purposes
+	if (!environmentService.isBuilt && !environmentService.disableTelemetry) {
+		return true;
+	}
 	return !(environmentService.disableTelemetry || !productService.enableTelemetry || environmentService.extensionTestsLocationURI);
 }
 
+/**
+ * Checks if we're only logging to the output channels for debug purposes and telemetry is not actually being sent
+ * @param productService
+ * @param environmentService
+ * @returns True if telemetry is actually disabled and we're only logging for debug purposes
+ */
+export function isLoggingOnly(productService: IProductService, environmentService: IEnvironmentService): boolean {
+	// Logging only mode is only for OSS
+	if (environmentService.isBuilt) {
+		return false;
+	}
+
+	if (environmentService.disableTelemetry) {
+		return false;
```

Comment:
If telemetry is disabled, shouldn't `isLoggingOnly` be true, i.e. we're only logging and not actually sending telemetry?

Reply:
No,

If telemetry is disabled via `--disableTelemetry` the log shouldn't have anything at all. The goal here is to have it when you're in OSS you can see telemetry flowing in the output channel the same way it would be sent. So all cases where telemetry is off it should not be getting logged either. I definitely understand how the logic can be confusing because we're kinda doing some strange things here in attempt to make it so that telemetry is easier to debug. Let me know if you have any ideas on how to clear that up. You can see that this check is basically saying if telemetry is on + we're in OSS + we don't have telemetry actually sending via the product.json containing the telemetry key then all we're doing is debug logging.

Reply:
This is a complete nit but maybe a comment to that effect would help, or `isLoggingOnly` -> `isOutputLoggingEnabled` or `shouldLogTelemetryOutput`? From just scanning the function docstring `True if telemetry is actually disabled and we're only logging for debug purposes` this line tripped me up 🫠

Reply:
Update comment